staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/structuredmerge.go

```diff
@@ -41,31 +41,31 @@ var _ Manager = &structuredMergeManager{}
 
 // NewStructuredMergeManager creates a new Manager that merges apply requests
 // and update managed fields for other types of requests.
-func NewStructuredMergeManager(typeConverter internal.TypeConverter, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, gv schema.GroupVersion, hub schema.GroupVersion) (Manager, error) {
+func NewStructuredMergeManager(typeConverter TypeConverter, objectConverter runtime.ObjectConvertor, objectDefaulter runtime.ObjectDefaulter, gv schema.GroupVersion, hub schema.GroupVersion) (Manager, error) {
 	return &structuredMergeManager{
 		typeConverter:   typeConverter,
 		objectConverter: objectConverter,
 		objectDefaulter: objectDefaulter,
 		groupVersion:    gv,
 		hubVersion:      hub,
 		updater: merge.Updater{
-			Converter: internal.NewVersionConverter(typeConverter, objectConverter, hub), // This is the converter provided to SMD from k8s
+			Converter: NewVersionConverter(typeConverter, objectConverter, hub), // This is the converter provided to SMD from k8s
```

Comment:
If you're moving the version converter from internal to here, you'll have to make it private (uncapitalize) everything in that package.

Reply:
That would be ideal, but would it cause problems with the versionconverter test file since part of a different package?

Calling `vc := fieldmanager.NewVersionConverter(tc, oc, schema.GroupVersion{Group: "apps", Version: runtime.APIVersionInternal})` in `versionconverter_test.go` would be problematic if the function was private right?

Reply:
you can change the test "package" to remove the `_test` to make it an "internal test" and avoid that problem ;-)

Reply:
Updated. The `fieldmanager_test.go` imported the version converter so that was changed to an internal test as well and a couple other packages depended on that so they were changed to internal tests too.